<a href="https://colab.research.google.com/github/disiree31-abel/cs82a-portfolio/blob/main/module3_lab3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd

df = pd.read_csv("bridges.csv")

print("Shape:", df.shape)
display(df.head())

print("\nData types:")
print(df.dtypes)

Shape: (108, 13)


,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
0,E1,M,3.0,1818,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
1,E2,A,25.0,1819,HIGHWAY,1037.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
2,E3,A,39.0,1829,AQUEDUCT,NaN,1.0,N,THROUGH,WOOD,NaN,S,WOOD
3,E5,A,29.0,1837,HIGHWAY,1000.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
4,E6,M,23.0,1838,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,NaN,S,WOOD



Data types:
Id           object
river        object
location    float64
erected       int64
purpose      object
length      float64
lanes       float64
clear-g      object
t-or-d       object
material     object
span         object
rel-l        object
type         object
dtype: object


In [3]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
location,107.0,25.978505,13.665871,1.0,15.5,27.0,37.5,52.0
erected,108.0,1905.314815,37.173664,1818.0,1884.0,1903.0,1928.0,1986.0
length,81.0,1567.469136,747.491523,804.0,1000.0,1300.0,2000.0,4558.0
lanes,92.0,2.630435,1.164657,1.0,2.0,2.0,4.0,6.0


Plausibility Assumption

I am assuming that a bridge length between 500 and 4,000 feet is a reasonable range for this dataset. I will flag anything outside this range so I can take a closer look at it. I chose this range because most of the lengths in the dataset are between about 800 and 2,000 feet, so a length over 4,000 feet stands out from the rest.

In [4]:
flagged = df[(df["length"] < 500) | (df["length"] > 4000)]

display(flagged)

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
31,E34,O,41.0,1888,RR,4558.0,2.0,G,THROUGH,STEEL,LONG,F,SIMPLE-T


In [6]:
print("Number of flagged rows:", len(flagged))

Number of flagged rows: 1


In [7]:
print("Flagged lengths:")
print(flagged["length"].tolist())

Flagged lengths:
[4558.0]


In [8]:
display(flagged)

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
31,E34,O,41.0,1888,RR,4558.0,2.0,G,THROUGH,STEEL,LONG,F,SIMPLE-T


Flagged Row and Decision

The screening found one bridge with a length of 4,558 feet. I flagged it because it is above the 4,000-foot limit I chose and is much longer than most of the bridges in the dataset.

I would keep this value for now because a bridge that is 4,558 feet long is unusual but still possible. I would check the original source to make sure the value was entered correctly before deciding whether it should be changed or removed.

In [9]:
display(flagged)

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
31,E34,O,41.0,1888,RR,4558.0,2.0,G,THROUGH,STEEL,LONG,F,SIMPLE-T
